logger/logger_metrics_interface.cc

```diff
@@ -32,79 +32,107 @@ string getEnv(const char * variable)
 /* LOGGER METRICS                                                           */
 /****************************************************************************/
 
-string ILoggerMetrics::parentObjectId = "";
-shared_ptr<ILoggerMetrics> logger;
 const string ILoggerMetrics::METRICS = "metrics";
 const string ILoggerMetrics::PROCESS = "process";
 const string ILoggerMetrics::META = "meta";
-bool ILoggerMetrics::failSafe;
+
+bool ILoggerMetrics::failSafe(false);
```

Comment:
"true" would be better. By default we don't want the logger metrics to get in the way. Should logging fail, the default behaviour is to avoid crashing the calling app.